### ISSUE TITLE
Persistable "load" will always pass as Successful?

### ISSUE THREAD

**Issue by Kithio** [wurstscript/WurstStdlib2#48]
It seems that as long as you have LocalFiles enabled, the "load" function will never give back "success = false" to it's callback.

Personally I find it confusing to have it this way because trying to load invalid data would still cause it to give a "successful" response.... Why success is being passed to the "onLoaded" event when it's never going to be false as long as the player has LocalFiles enabled is beyond me... Especially since the example  shown using Persistable to save/load data uses network to check if local files are enabled BEFORE calling the load function to begin with.

**Comment by Frotty** [comment]
Yea, it should probably pass `success and deserializeSuccess`.
Looks like a regression from adding LocalFile handling.
I will wait for @SamuelMoriarty to respond and then apply a fix as needed.
Thanks for the report.

**Comment by Kithio** [comment]
I just found it very strange since it means there's no way to actually find out if they tried to load a valid file or not :P

**Comment by SamuelMoriarty** [comment]
Yes, it would make sense for them both to be passed.
They should also probably be renamed `fileParseSuccess` and `deserializeSuccess`.
There are three stages where loading can potentially fail:
1. Local User can't read files because LocalFiles aren't enabled,
2. Local User couldn't decode the data in the file due to corruption, i.e. StringBuffer failed
3. The data is decoded correctly, but there is a logical error in the data itself, i.e. deserialize method failed

`success` corresponds to 1 and 2, `deserializeSuccess` corresponds to 3.
In either scenario, the persistable class supports supplying default values to the class in case of an error.

**Comment by Kithio** [comment]
I would like a way to actually know if the file that attempted being loaded actually existed in the first place so I can report an error to invalid file or invalid data, if that is even possible.

But fixing the "success" thing would give me more of an idea about where things went wrong.

**Comment by SamuelMoriarty** [comment]
It's definitely a good idea, and it gave me a huge headache back when I was trying to get the initial version out. Admittedly I was somewhat in a hurry, so there's still lots of room for improvement.

The thing is, though, there's no easy way to test if the user doesn't have the file without actually trying to read it first, which is an expensive operation. Either way, whether the user has no data or it is corrupted - the outcome is one and the same, you have to ditch the old data (if any) and rewrite it. Hence the `.supplyDefault()` method being called in both scenarios.

But there definitely needs to be better error handling.

**Comment by Frotty** [comment]
I would suggest replacing the boolean with an enum that contains the possible failure cases, e.g.
```
enum LoadStatus
    FAIL_NOT_ENABLED
    FAIL_FILE_CORRUPT
    FAIL_DECODE
    SUCCESS
```

**Comment by SamuelMoriarty** [comment]
#49 Fixes this.

**Comment by Frotty** [comment]
fixed by #49
